### PR TITLE
Libvirt support for bootstrap + etcd phases

### PR DESCRIPTION
**Pull request by squeed**
This introduces basic libvirt support for the following steps:
* Assets
* Topology
* Bootstrap
* Etcd
* tnc_dns

### CODE REVIEW

**Comment by squeed** [comment]
ok to test

**Comment by crawford** [comment]
I tried running this, but it ultimately failed while trying to create AWS resources (while realizing `data.terraform_remote_state.bootstrap`).

File: modules/libvirt/volume/main.tf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+variable "coreos_qow_path" {
```
[the remark this answers is not in the document]
Comment (squeed) [reply modules/libvirt/volume/main.tf]: Yes. Yes they are.

**Comment by squeed** [comment]
@crawford I'm not sure where that's happening. Right now, we only support the bootstrap phase, so you have to manually run 
```
tectonic install assets
tectonic install bootstrap
```

I tried this without a valid AWS configuration and it worked.

**Comment by squeed** [comment]
retest this please

**Comment by squeed** [comment]
all green! don't merge until the operator changes are merged.

**Comment by enxebre** [comment]
retest this please

**Comment by squeed** [comment]
Failing on container linux version - should I skip this test?

**Comment by squat** [comment]
@enxebre I thought you had disabled the check CL version test, no?

**Comment by enxebre** [comment]
@squat @squeed the test disabled is for the cl channel, the cl version should succeed, this probably needs to be updated as this is changing the paths https://github.com/coreos/tectonic-installer/blob/master/tests/rspec/lib/aws_cluster.rb#L269

**Comment by squeed** [comment]
@enxebre ah, good find! let's see if this does it.

**Comment by squeed** [comment]
rebased.
The TNC changes have made master, so this can be merged on green.

**Comment by squeed** [comment]
The tests are failing, and I don't know how they ever worked in UT2: it's trying to get the worker instance ids from the bootstrap.tfstate file. Now that we have split tfstates...
Thoughts, @spangenberg?

**Comment by enxebre** [comment]
@squeed tests are failing because they can't determine the bootkube service has run successfully because they can't ssh into the machines. Have you deployed this locally successfully? I'll try to give it a go

**Comment by squeed** [comment]
It's red because I typo'd the tnc image name. Let's try this again.

**Comment by squeed** [comment]
All green! ready to merge for real, this time.

**Pull request merged**